pkg/image/apis/image/helper.go

```diff
@@ -145,7 +144,7 @@ func IsRegistryDockerHub(registry string) bool {
 func ParseDockerImageReference(spec string) (DockerImageReference, error) {
```

Comment:
@deads2k We can have it in the versioned API because registry also uses `DockerImageReference` ?

Reply:
> @deads2k We can have it in the versioned API because registry also uses DockerImageReference ?

Point me to a call-site just so I can see for sure.  Helpers have been a problem both upstream and down.

Reply:
@deads2k https://github.com/openshift/origin/blob/d1fcf3872eb13b399c7912e09ed5c4a66dd3c541/pkg/dockerregistry/server/pullthroughmanifestservice.go#L46 https://github.com/openshift/origin/blob/d1fcf3872eb13b399c7912e09ed5c4a66dd3c541/pkg/dockerregistry/server/signaturedispatcher.go#L73 and some more.

Reply:
@deads2k Thanks for ignoring my question without explanation.

Reply:
> @deads2k Thanks for ignoring my question without explanation.

Sorry it looked like that.  I had a comment that must not have pushed send on.

API helpers for versioned and unversioned types are not a solved problem.  Drift between them causes problems and keeping them coupled with conversions is slow.  This pull isn't trying to solve that problem and doesn't make it appreciably worse.

Reply:
@deads2k but DockerImageReference are not supposed to be a part of our API and not supposed be versioned. Why can't we just move it to a separate package? We haven't got the parser in the external API yet, so we don't restricted by backward compatibility.